tests/integration/docker/test_docker.py

```diff
@@ -0,0 +1,399 @@
+import logging
+from typing import NamedTuple
+
+import pytest
+
+from localstack import config
+from localstack.utils.bootstrap import PortMappings
+from localstack.utils.common import short_uid
+from localstack.utils.docker import CmdDockerClient as DockerClient
+from localstack.utils.docker import (
+    ContainerException,
+    DockerContainerStatus,
+    NoSuchContainer,
+    NoSuchImage,
+    Util,
+)
+
+ContainerInfo = NamedTuple(
+    "ContainerInfo",
+    [
+        ("container_id", str),
+        ("container_name", str),
+    ],
+)
+
+LOG = logging.getLogger(__name__)
+
+container_name_prefix = "lst_test_"
+
+
+def _random_container_name() -> str:
+    return f"{container_name_prefix}{short_uid()}"
+
+
+@pytest.fixture
+def dummy_container(create_container):
+    """Returns a container that is created but not started"""
+    return create_container("alpine", command=["sh", "-c", "while true; do sleep 1; done"])
+
+
+@pytest.fixture
+def create_container(docker_client: DockerClient):
+    """
+    Uses the factory as fixture pattern to wrap DockerClient.create_container as a factory that
+    removes the containers after the fixture is cleaned up.
+    """
+    containers = list()
+
+    def _create_container(image_name: str, **kwargs):
+        kwargs["name"] = kwargs.get("name", _random_container_name())
+        cid = docker_client.create_container(image_name, **kwargs)
+        cid = cid.strip()
+        containers.append(cid)
+        return ContainerInfo(cid, kwargs["name"])  # FIXME name should come from docker_client
+
+    yield _create_container
+
+    for c in containers:
+        try:
+            docker_client.remove_container(c)
+        except Exception:
+            LOG.warning("failed to remove test container %s", c)
+
+
+class TestDockerClient:
```

Comment:
Out of curiosity: I'm assuming we're not extending from `unittest.TestCase` here, to be able to use the `pytest.fixture`s in the test? (or could we combine both?)

Reply:
the pytest fixtures only work with unittest TestCase classes with silly workarounds/annotations. there's no value in adding `unittest.TestCase` as a base class in my opinion.

Reply:
technically even the class is unnecessary, but it groups the tests nicely :-)